### PR TITLE
[RFC] Extend Threat Fieldset - Stage 3 Proposal

### PR DESCRIPTION
**Pull request by peasead**
- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)? Yes
- Have you followed the [contributor guidelines](https://github.com/elastic/ecs/blob/master/CONTRIBUTING.md)? Yes
- For proposing substantial changes or additions to the schema, have you reviewed the [RFC process](https://github.com/elastic/ecs/blob/master/rfcs/README.md)? Yes
- If submitting code/script changes, have you verified all tests pass locally using `make test`? Yes
- If submitting schema/fields updates, have you generated new artifacts by running `make` and committed those changes? Yes
- Is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed. Yes
- Have you added an entry to the [CHANGELOG.next.md](https://github.com/elastic/ecs/blob/master/CHANGELOG.next.md)? NA

### CODE REVIEW

File: rfcs/text/0018-extend-threat-group-software.md [hunk 1]
```diff
@@ -1,7 +1,8 @@
 # 0018: Extend Threat Fieldset
 <!-- Leave this ID at 0000. The ECS team will assign a unique, contiguous RFC number upon merging the initial stage of this RFC. -->
 
-- Stage: **2 (candidate)** <!-- Update to reflect target stage. See https://elastic.github.io/ecs/stages.html -->
+
+- Stage: **3 (candidate)** <!-- Update to reflect target stage. See https://elastic.github.io/ecs/stages.html -->
```
Comment (ebeahan) [inline rfcs/text/0018-extend-threat-group-software.md:5]:
```suggestion
- Stage: **3 (finished)** <!-- Update to reflect target stage. See https://elastic.github.io/ecs/stages.html -->
```

File: rfcs/text/0018-extend-threat-group-software.md [hunk 2]
```diff
@@ -61,20 +62,26 @@ Currently, tactic, technique, and sub-techniques are also included in rules for
...
     "threat.technique.subtechnique.id": "T1087.002",
     "threat.technique.subtechnique.name": "Domain Account",
-    "threat.technique.subtechnique.reference": "https://attack.mitre.org/techniques/T1087/002/",
-
-// New Software fields
+    "threat.technique.subtechnique.reference": "https://attack.mitre.org/techniques/T1087/002/"
+}
+```
+**New Software fields**
+```json
```
Comment (ebeahan) [inline rfcs/text/0018-extend-threat-group-software.md:69]:
```suggestion
```json
{
```

Looked like a missing brace as I was reviewing.

File: rfcs/text/0018-extend-threat-group-software.md [hunk 3]
```diff
@@ -61,20 +62,26 @@ Currently, tactic, technique, and sub-techniques are also included in rules for
...
     },
     "threat.software.reference": "https://attack.mitre.org/software/S0552/",
-    "threat.software.type": "Tool",
-
-// New Group fields
+    "threat.software.type": "Tool"
+}
+```
+**New Group fields**
+```json
```
Comment (ebeahan) [inline rfcs/text/0018-extend-threat-group-software.md:84]:
```suggestion
```json
{
```

Same here re: the open brace.

**Review by ebeahan — COMMENTED** [review]
Looks great!

@peasead Can you add a sentence or two under `Concerns` capturing the resolution for the one concern (fields align with MITRE ATT&CK but are not limited to)?

**Comment by peasead** [comment]
> Looks great!
> 
> 
> 
> @peasead Can you add a sentence or two under `Concerns` capturing the resolution for the one concern (fields align with MITRE ATT&CK but are not limited to)? 

Done!

**Comment by peasead** [comment]
Just checking in on this PR.

**Review by ebeahan — COMMENTED** [review]
Looks great! Sorry for the delay in revisiting.

One item I wanted to confirm: is `threat.software.alias` a new addition?

**Comment by peasead** [comment]
> Looks great! Sorry for the delay in revisiting.
> 
> One item I wanted to confirm: is `threat.software.alias` a new addition?

Yes, I was researching something else and realized that there was software that had multiple names, similar to groups that have multiple names. Looking at this, I changed the examples to use groups and software with alias'.

**Review by devonakerr — APPROVED** [review]
Hey, I think we might be good!

**Comment by ebeahan** [comment]
> Yes, I was researching something else and realized that there was software that had multiple names, similar to groups that have multiple names. Looking at this, I changed the examples to use groups and software with alias'.

Thanks for clarifying, @peasead! I will do a final read-through today.

**Review by ebeahan — APPROVED** [review]

**Pull request merged**